src/agent/onefuzz-agent/src/tasks/heartbeat.rs

```diff
@@ -27,114 +24,52 @@ pub enum HeartbeatData {
 }
 
 #[derive(Debug, Deserialize, Serialize, Clone)]
-struct Heartbeat<'a> {
-    task_id: Uuid,
+struct Heartbeat {
+    task_id: Option<Uuid>,
     machine_id: Uuid,
-    machine_name: &'a str,
+    machine_name: String,
     data: Vec<HeartbeatData>,
 }
 
-pub struct HeartbeatClient {
-    cancelled: Arc<Notify>,
-    messages: Arc<Mutex<HashSet<HeartbeatData>>>,
-    _heartbeat_process: JoinHandle<Result<()>>,
-}
-
-impl Drop for HeartbeatClient {
-    fn drop(&mut self) {
-        self.cancelled.notify();
-    }
-}
-
-impl HeartbeatClient {
-    pub fn init(queue_url: Url, task_id: Uuid) -> Self {
-        HeartbeatClient::init_with_period(queue_url, task_id, DEFAULT_HEARTBEAT_PERIOD)
-    }
+pub type TaskHeartbeatClient = HeartbeatClient<HeartbeatData>;
 
-    pub fn init_with_period(queue_url: Url, task_id: Uuid, heartbeat_period: Duration) -> Self {
-        let messages = Arc::new(Mutex::new(HashSet::new()));
-        let cancelled = Arc::new(Notify::new());
-        let _heartbeat_process = HeartbeatClient::start_background_process(
-            task_id,
+pub fn init_task_heartbeat(
```

Comment:
I think it make the most sense to have this be an additional constructor for `HeartbeatClient`, and make it shareable between the agent and supervisor.